### ISSUE TITLE
[Request] Please add UI support Inventory_Cargo and FreighterInventory_Cargo (maybe *_Cargo)

### ISSUE THREAD

**Issue by SplinterGU** [goatfungus/NMSSaveEditor#831]
please, add ui support to:

PlayerStateData.Inventory_Cargo
PlayerStateData.FreitherInventory_Cargo
ShipOwnership.*.Inventory_Cargo

(maybe other *_Cargo)

this inventary are used as extra cargo... in the game this is access only with "quick transfer", but is very annoying...

![image](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/9f618844-4ba0-4d93-97c6-3d0cf06b7316)
![image](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/aa2abfa0-60e0-4a52-95fe-2df0336d5d1a)
![image](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/ad30038c-5556-4b23-a967-20fee99295c9)

**Comment by MGJerry** [comment]
Wait, hol' up. How are you even accessing those inventories? I thought those were deprecated a few updates ago? Those items should have gone into your now combined inventory the moment you open the inventory ingame.

**Comment by SplinterGU** [comment]
> Wait, hol' up. How are you even accessing those inventories? I thought those were deprecated a few updates ago? Those items should have gone into your now combined inventory the moment you open the inventory ingame.
> 
> Edit: Just opened an old backup in the latest version of the editor, those Cargo inv are still accessible if the editor see an older game version (from when those Cargo inv are still in use) ![image](https://user-images.githubusercontent.com/85227828/243635200-53f065af-b353-49a9-bf78-133c3cef8399.png)

cargo is normal inventory in my editor

![image](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/76ef2980-0b6f-4d82-99cf-7539d9151cf9)

**Comment by MGJerry** [comment]
Ah, that... Yeah, since NMS 4.0, General and Cargo inventories got merged, and all data is stored in the Inventory names (no "_Cargo") and those "_Cargo" inventories are now deprecated ingame. @goatfungus probably choose "Cargo" 'cause the new stack limits are similar to the ones of "Cargo"s, but I still feel they should be named "General"s to avoid confusions.

Side notes, about the data in Cargo inventories - yeah... some of them still have data leftover in some of my save files, they don't do anything (except if you count increasing the size of the save then yes, they do, by like a bit)

**Comment by SplinterGU** [comment]
> Ah, that... Yeah, since NMS 4.0, General and Cargo inventories got merged, and all data is stored in the Inventory names (no "_Cargo") and those "_Cargo" inventories are now deprecated ingame. @goatfungus probably choose "Cargo" 'cause the new stack limits are similar to the ones of "Cargo"s, but I still feel they should be named "General"s to avoid confusions.
> 
> Side notes, about the data in Cargo inventories - yeah... some of them still have data leftover in some of my save files, they don't do anything (except if you count increasing the size of the save then yes, they do, by like a bit)

no, no... this inventary is functional... I remove all items from it... then when main inventary is full, the items will go to the inventary_cargo... this inventary works... don't is obsolete...

**Comment by SplinterGU** [comment]
look the images... yellow are normal inventory... red is inventory_cargo... it is functional... this is a new ship from some days ago... maybe v4.25... this isn't obsolete...

![275850_20230606132452_1](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/e105e8fb-4ab4-4a76-89a0-60ebc730f629)
![275850_20230606132348_1](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/da21acf2-9d51-43fe-b27b-14c4beae61b5)
![275850_20230606132310_1](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/a9a5b4cf-e915-41bc-8b85-0c6dfc88e4df)
![275850_20230606132235_1](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/5bf34ec3-c7b7-4c73-8351-16e32c6ff60b)
![275850_20230606132201_1](https://github.com/goatfungus/NMSSaveEditor/assets/95715426/4d6f53e2-dff7-4693-a0c4-c6cbda1acdd0)

**Comment by MGJerry** [comment]
Ok, I kind of get the gist of it now. After testing with 3 of my vanilla saves (1 in Normal difficulty, 1 in Survival difficulty and 1 in Expedition 7), none of them ever overflow items to the unused Cargo space, and instead overflow to the ships or freighter (if the player can access them). So, the fact that those items go there mean this is a strictly "modded" problem. Don't know if goat gonna support modded stuff like this, so I'm gonna leave it to him to respond to this.

If you really need to modify those Cargo inventories, the best I can offer is to used the 1.13.8 version of the editor (when those haven't been taken out yet)
Here's the link to it: https://github.com/goatfungus/NMSSaveEditor/blob/1cffdf4759bed90f00480f51f7d041000a8d956c/NMSSaveEditor.zip

![image](https://github.com/goatfungus/NMSSaveEditor/assets/85227828/ba7ddb38-e260-4797-a2c3-e7b4da8d6984)
The "==Unused==" tab is what you need for those inventories ending in "_Cargo"

**Comment by MGJerry** [comment]
> you can get an exosuit overflow if you get items from anomaly (transfers from other users) or if you use duplication glitch or something other method (I don't remember now), in the ship if you craft item from asteroids in the space...

I got a friend up to test this and yep, it does overflow when somebody gives you an item when inventories a full, so I do apologize on that part. Ngl, that's a massive oversight they should have fixed (although I can't find much posts about it, nor too much mentions about it in the discord server)

> but what about all bugs or missing features of this old version? I think that isn't a solution...

Just edit what you need in the v1.13.8 editor, copy what's in the raw json and paste it to the newest editor, that's literally my best suggestion for you right now. And for the re-adding of the "_Cargo" tabs, we will have to wait for Brendon's response (or you know, until HelloGames actually fix their mistake)

**Comment by SplinterGU** [comment]
No problem...:)
About edit json raw, isn't easy... each slot have coords... Isn't easy...
I did several tools for mánage nms json raw data, but isn't the idea...
I don't want do double work... I don't want do a new editor... This editor
Is great...
Anyway, thanks!
